app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/notification/NotificationResponse.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.models.notification
+
+import play.api.libs.json._
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.GenericErrorResponse
+
+import java.time.Instant
+
+sealed trait NotificationResponse
+
+object NotificationResponse {
+
+  final case class SuccessPushNotificationResponse(notificationId: String) extends NotificationResponse
+
+  object SuccessPushNotificationResponse {
+    implicit val format: OFormat[SuccessPushNotificationResponse] = Json.format[SuccessPushNotificationResponse]
+  }
+
+  final case class SuccessBoxNotificationResponse(boxId: String) extends NotificationResponse
+
+  object SuccessBoxNotificationResponse {
+    implicit val format: OFormat[SuccessBoxNotificationResponse] = Json.format[SuccessBoxNotificationResponse]
+  }
+
+  case class FailedBoxIdNotificationResponse(dateTime: Instant, debugMessage: String) extends GenericErrorResponse {
+    override val message: String = "Box Id error"
+  }
+
+  object FailedBoxIdNotificationResponse {
+
+    implicit val read: Reads[FailedBoxIdNotificationResponse] =
+      Json.reads[FailedBoxIdNotificationResponse]
+
+    implicit var write: OWrites[FailedBoxIdNotificationResponse] = {
+
+      import play.api.libs.functional.syntax._
+
+      (
+        (__ \ "dateTime").write[Instant] and
+          (__ \ "message").write[String] and
+          (__ \ "debugMessage").write[String]
+        ) (r => (r.dateTime, r.message, r.debugMessage))
+    }
+  }
```

Comment:
Why do we need explicit reads and writes for error responses but we only have a format for the success responses. Can we not just get away with the format for both? We only tend to have formats for other JSON conversions in our repository so far. This is the first use of Reads and OWrites so just curious why they are needed here

Reply:
This is needed because the Implicit OFormat is not able to write override member. I tried a new designed here. I have an specialised notification that derived from GenericErrorResponse, which override message. It is this message that the default implicit cannot write (I guess because the apply of the specialised one only takes datetime and debugMessage) so we need a specialised one. The concept is that I tried to create specialised errorResponse where the message is a constant (not passed in), and so we will have different one for Notification and future maybe also for draftMovement, Submission, etc. So we do not need to pass message in. Chris, Nicole and I had a chat the other day about this, but I would also appreciated your suggestion.

Reply:
This GenericErrorResponse stuff probably ties into our general error response approach discussion we need to have?

Reply:
Apparently there is a library that allows you to define OFormats for sealed traits easier:
https://github.com/julienrf/play-json-derived-codecs

This is being used by other projects in the HMRC github estate:
https://github.com/search?q=org%3Ahmrc+%2F%28%3F-i%29play-json-derived-codecs%2F&type=code

Reply:
At the moment I do not see the need to import an extra dependency, when we can do this with Scala Play-Framework relatively in a simple way and at the moment this is only in one place. I don`t think at this stage this should block a PR for being merged. I could look into in the future as the push notification still ongoing.